dbdimp.c

```diff
@@ -582,13 +582,9 @@ char **fill_out_embedded_options(PerlIO *stream,
   char c;
   char *ptr;
   char **options_list= NULL;
+  dTHX;
 
-  if (!(options_list= (char **) calloc(cnt, sizeof(char *))))
-  {
-    PerlIO_printf(stream,
-                  "Initialize embedded server. Out of memory \n");
-    return NULL;
-  }
+  Newz(908, options_list, cnt, char *);
```

Comment:
one concern, so now it's just `croak` when OOM and not output anything to stream :)

Reply:
Yes, when there is no free memory, other memory allocations (in perl interpreter too; in DBI.xs too) would fail too.